### ISSUE TITLE
Bug: Flat config ignores option dont work

### ISSUE THREAD

**Issue by JuanDa237** [eslint/eslint#18446]
### Environment

Node version: 
20.11.0
npm version: 
10.2.4
Local ESLint version:
9.2.0
Global ESLint version: 
9.2.0
Operating System: 
Windows 10

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
// eslint.config.mjs
import pluginJs from "@eslint/js";
import eslintConfigPrettier from "eslint-config-prettier";
import pluginReactConfig from "eslint-plugin-react/configs/recommended.js";
import globals from "globals";
import tsEslint from "typescript-eslint";

/** @type { import("eslint").Linter.FlatConfig[] } */
export default [
  {
    files: ["**/*.js"],
    languageOptions: { sourceType: "script" },
    ignores: [".vite/", "postcss.config.js", "tailwind.config.js"],
  },
  { languageOptions: { globals: globals.browser } },
  pluginJs.configs.recommended,
  ...tsEslint.configs.recommended,
  pluginReactConfig,
  eslintConfigPrettier,
  {
    rules: {
      "react/react-in-jsx-scope": "off",
      "react/prop-types": "off",
      "no-unused-vars": "off",
      "@typescript-eslint/no-unused-vars": "off",
    },
  },
];
```
The important part is the ignores option at first obj

### What did you expect to happen?

I expect it to ignore the ./vite folder at root, and ignore the tailwind and postcss configurations.

![image](https://github.com/eslint/eslint/assets/63322751/f4f12e4d-58c3-41eb-a0db-2ee12e5bf04e)

I have also tried with:

ignores: [".vite/", "**/**.config.*"]

And it did not work

### What actually happened?

whit running `npx eslint .` i got the files [".vite/", "postcss.config.js", "tailwind.config.js"], evaluated

### Link to Minimal Reproducible Example

none.for/now

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I did wrong the configuration, or it is actully bug?

**Comment by aladdin-add** [comment]
it must have no other keys than `ignores`, see https://eslint.org/docs/latest/use/configure/ignore#ignoring-files

**Comment by mheinzerling** [comment]
@aladdin-add Unfortunately this is also true for 'name'. Is this by purpose? Should I create a ticket to fix this? (A name would be nice for the config inspector)

**Comment by aladdin-add** [comment]
> Unfortunately this is also true for 'name'.

not sure I'm following.  the `name` key was only used in the error message (there was an issue but should have been fixed: https://github.com/humanwhocodes/config-array/pull/131 ,https://github.com/humanwhocodes/config-array/pull/133)

**Comment by mdjermanovic** [comment]
Yeah, the problem with `name` in global ignores has been fixed in ESLint v9.0.0. Maybe you're using ESLint v8.57.0?

**Comment by mheinzerling** [comment]
Good to know. Still need to get the flat config running in 8 before updating to 9. Thanks.
